### ISSUE TITLE
Image classification with rotations

### ISSUE THREAD

**Issue by JohnCraigPublic** [Microsoft/CNTK#1748]
I have an image classifier working well in CNTK.   In my training set of images there is a "right side up" orientation for each image.  All samples in the training and test sets are in this orientation.  As a part of classification, I want to also determine if the image I am evaluating is rotated 0, 90, -90, or 180 relative the the train set orientation.   So, when I go to eval an image, I rotate it to all 4 possible orientations, and then hand these 4 images to my evaluator.   So, I call my Eval 4 times, and it is using Softmax, so I get a probability back from each call.  So, usually, this works: the highest probability from any of the 4 calls will tell me the image class and its orientation.  However, it is sometimes wrong, and sometimes, the 2nd best probability comes *very* close to matching/exceeding the correct one.  So, I am thinking that what I should better do is this: If I have N classes, rather than computing 4 Softmaxes of size N, I should rather get numeric output (not Softmax) from each of the 4 evals, and then put them all together thru a Softmax of size 4*N.   Do you agree?  This should work better?   Further, I am using BrainScript and evaluating with C++ and I'd prefer to continue that way, so don't tell me to write come python code.  Rather, it seems I would need to write my own Softmax function in C++ to do this.  I think I've seen the code (or formula) for how to compute Softmax, but a pointer to it would be appreciated.  Or is there some other way to classify with 4-fold rotations (e.g. I could do data augmentation and make my training set 4X bigger by feeding in all images rotated all ways -- but that seems somehow silly -- but maybe....?).  Thanks for your deep thoughts.

**Comment by JohnCraigPublic** [comment]
Asked this on StackOverflow.....

**Comment by JohnCraigPublic** [comment]
Nobody commented there either....
